docs/notes/ml.rst

```diff
@@ -1,11 +1,116 @@
 Machine Learning with Vineyard
 ==============================
+**Vineyard-ML**: A Vineyard package that integrates Machine Learning Frameworks to Vineyard.
 
 TensorFlow
 ----------
+Using Numpy Data
+^^^^^^^^^^^^^^^^
+
+.. code:: python
+
+     >>> import tensorflow as tf
+     >>> from vineyard.contrib.ml import tensorflow
+     >>> dataset = tf.data.Dataset.from_tensor_slices((data, label))
+     >>> data_id = client.put(dataset)
+     >>> vin_data = client.get(data_id)
+     
+The ``vin_data`` will be a shared-memory object from the vineyard.  
+
+Using Dataframe
+^^^^^^^^^^^^^^^
+.. code:: python
```

Comment:
Leave a blank line between blocks in the text, e.g., a blank is required between the title line and the `.. code:: python` line.